components/tor/tor_launcher_factory.cc

```diff
@@ -206,7 +195,9 @@ void TorLauncherFactory::OnTorLaunched(bool result, int64_t pid) {
   }
   for (auto& observer : observers_)
     observer.OnTorLaunched(result, pid);
-  control_->Start();
+  tor_file_watcher_->StartWatching(
+      base::BindOnce(&TorLauncherFactory::OnTorControlPrerequisitesReady,
```

Comment:
please add thread checks to any methods that are called with base::Bind

Reply:
fixed in e6fa00fad62e4ad671e59f9675a1abdaab11d919